tests/scripts/local_monitoring_tests/rest_apis/test_bulkAPI.py

```diff
@@ -221,3 +229,197 @@ def test_bulk_validate_datasource_missing(cluster_type):
+        # Step 2: Poll GET Bulk API until metadata is available
+        # --------------------------------------------------
+        metadata = None
+
+        for _ in range(10):
+            get_response = get_bulk_job_status(
+                job_id,
+                include="metadata",
+                logger=logging
+            )
+
+            assert get_response.status_code == SUCCESS_200_STATUS_CODE
+            result = get_response.json()
+
+            metadata = result.get("metadata", {})
+            if metadata.get("datasources"):
+                break
+
+            logger.info("Bulk job still processing, retrying...")
+            time.sleep(5)
+
+        assert metadata and metadata.get("datasources"), \
+            "Bulk job did not return metadata within expected time"
+
+        datasources = metadata["datasources"]
+
+        # --------------------------------------------------
+        # Step 3: Expected filters
+        # --------------------------------------------------
+        expected_namespace = payload["filter"]["include"]["namespace"][0]
+        expected_workload = payload["filter"]["include"]["workload"][0]
+        expected_container = payload["filter"]["include"]["containers"][0]
+
+        # --------------------------------------------------
+        # Step 4: Validate metadata hierarchy
+        # --------------------------------------------------
+        for ds_name, ds_data in datasources.items():
+            clusters = ds_data.get("clusters", {})
+            assert clusters, f"No clusters found in datasource {ds_name}"
+
+            for cluster_name, cluster_data in clusters.items():
+                namespaces = cluster_data.get("namespaces", {})
+                assert namespaces, "No namespaces found in cluster metadata"
+
+                # Namespace filter
+                assert list(namespaces.keys()) == [expected_namespace], \
+                    f"Unexpected namespaces: {list(namespaces.keys())}"
+
+                ns_data = namespaces[expected_namespace]
+                workloads = ns_data.get("workloads", {})
+                assert workloads, "No workloads found under namespace"
+
+                # Workload filter
+                assert list(workloads.keys()) == [expected_workload], \
+                    f"Unexpected workloads: {list(workloads.keys())}"
+
+                wl_data = workloads[expected_workload]
+                containers = wl_data.get("containers", {})
+                assert containers, "No containers found under workload"
+
+                # Container filter
+                assert list(containers.keys()) == [expected_container], \
+                    f"Unexpected containers: {list(containers.keys())}"
+
+        logger.info(
+            "Bulk API filter validation successful for namespace, workload, and container"
+        )
+
+@pytest.mark.test_bulk_api_ros
+@pytest.mark.sanity
+def test_bulk_api_namespace_only_filter(cluster_type, caplog):
```

Comment:
I think you need to follow [this](https://github.com/kruize/autotune/blob/1f870ac536f2eb424f4b0b0005327b42346ab9fc/tests/scripts/local_monitoring_tests/rest_apis/test_bulkAPI.py#L104) parametrize approach here to add multiple combinations. 

@chandrams can suggest better as well